src/id_allocator.rs

```diff
@@ -0,0 +1,201 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
+
+//! Provides allocation and releasing strategy for IDs.
+//! 
+//! This module implements an allocation strategies for all resource types
+//! that can be abstracted to an integer.
+
+
+//! Provides emulation for a minimal ARM PL031 Real Time Clock.
```

Comment:
Is this part needed?:)

Reply:
Nope it was a miss from my part. Thank you for pointing this out.